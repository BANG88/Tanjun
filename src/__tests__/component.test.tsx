import React from 'react'
import renderer from 'react-test-renderer'
import { Image, ScrollView, Text, View } from '../'

it('renders Text correctly', () => {
  const tree = renderer
    .create(
      <Text
        style={{
          marginTop: 60,
          fontSize: 40,
        }}>
        Tanjun Text
      </Text>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders View correctly', () => {
  const tree = renderer
    .create(
      <View
        style={{
          padding: 60,
          width: 300,
          height: 200,
        }}>
        <Text>Tanjun View</Text>
      </View>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Image correctly', () => {
  const tree = renderer
    .create(
      <Image
        source={{
          // Credit: unsplash.com
          uri: 'https://images.unsplash.com/photo-1528164344705-47542687000d?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2392&q=80',
        }}
        style={{
          width: 300,
          height: 300,
        }}>
        Tanjun
      </Image>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders ScrollView correctly', () => {
  const tree = renderer
    .create(
      <ScrollView
        style={{
          padding: 50,
        }}>
        <Image
          source={{
            // Credit: unsplash.com
            uri: 'https://images.unsplash.com/photo-1528164344705-47542687000d?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2392&q=80',
          }}
          style={{
            width: 300,
            height: 300,
          }}>
          Tanjun
        </Image>
      </ScrollView>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
