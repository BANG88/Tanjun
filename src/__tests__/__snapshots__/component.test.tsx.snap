// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Image correctly 1`] = `
<Image
  source={
    Object {
      "uri": "https://images.unsplash.com/photo-1528164344705-47542687000d?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2392&q=80",
    }
  }
  style={
    Object {
      "height": 209,
      "width": 209,
    }
  }
>
  Tanjun
</Image>
`;

exports[`renders ScrollView correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "padding": 35,
    }
  }
>
  <View>
    <Image
      source={
        Object {
          "uri": "https://images.unsplash.com/photo-1528164344705-47542687000d?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2392&q=80",
        }
      }
      style={
        Object {
          "height": 209,
          "width": 209,
        }
      }
    >
      Tanjun
    </Image>
  </View>
</RCTScrollView>
`;

exports[`renders Text correctly 1`] = `
<Text
  style={
    Object {
      "fontSize": 28,
      "marginTop": 42,
    }
  }
>
  Tanjun Text
</Text>
`;

exports[`renders View correctly 1`] = `
<View
  style={
    Object {
      "height": 139,
      "padding": 42,
      "width": 209,
    }
  }
>
  <Text>
    Tanjun View
  </Text>
</View>
`;
